ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/TimingProcessor.java

```diff
@@ -0,0 +1,47 @@
+package tech.pegasys.artemis.statetransition;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.common.primitives.UnsignedLong;
+import org.apache.logging.log4j.Level;
+import tech.pegasys.artemis.service.serviceutils.ServiceConfig;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.util.alogger.ALogger;
+
+import java.util.Date;
+
+import static tech.pegasys.artemis.datastructures.Constants.SECONDS_PER_SLOT;
+import static tech.pegasys.artemis.statetransition.util.ForkChoiceUtil.on_tick;
+
+public class TimingProcessor {
```

Comment:
Worst name ever.  I would have gone with...TimeController. 😜

Reply:
oh come on! I thought you'd love it. I'm heartbroken ...